src/UI/Buyer/src/app/shared/components/search/search.component.ts

```diff
@@ -56,27 +49,21 @@ export class SearchComponent implements OnInit, OnDestroy {
 
   private search() {
     const searchTerm = this.form.controls.search.value;
+    this.form.markAsPristine();
     // emit as undefined if empty string so sdk ignores parameter completely
     this.searched.emit(searchTerm || undefined);
   }
 
-  private onQueryParamChanges() {
-    // clear search bar if products are no longer filtered by search term
-    this.activatedRoute.queryParams
-      .pipe(
-        filter((queryParams) => typeof queryParams.search === 'undefined'),
-        takeWhile(() => this.alive)
-      )
-      .subscribe(() => {
-        this.form.controls['search'].setValue('');
-      });
-  }
-
   showClear(): boolean {
     return this.form.get('search').value !== '';
   }
 
   clear(): void {
+    this.form.markAsDirty();
+    this.form.setValue({ search: '' });
+  }
+
+  clearWithoutEmit(): void {
     this.form.setValue({ search: '' });
   }
```

Comment:
Doesn't look like clearWithoutEmit is being used?

Reply:
Its used in the other code you commented on ;)

Reply:
lol facepalm